lib/src/commands/admin.dart

```diff
@@ -1,112 +1,103 @@
 import 'package:nyxx/nyxx.dart';
 import 'package:nyxx_commands/nyxx_commands.dart';
-import 'package:running_on_dart/src/exception.dart';
-import 'package:running_on_dart/src/services/poop_name.dart';
+import 'package:running_on_dart/src/modules/poop_name.dart';
 
-ChatGroup admin = ChatGroup(
+final admin = ChatGroup(
   'admin',
   'Administrative commands',
   children: [
     ChatCommand(
       'cleanup',
       'Bulk deletes messages in a channel',
       id('admin-cleanup', (
-        IChatContext context,
-        @UseConverter(IntConverter(min: 1))
-        @Description('The number of messages to delete')
-        int count, [
-        @Description('The user from whom to delete messages') IUser? user,
+        ChatContext context,
+        @UseConverter(IntConverter(min: 1)) @Description('The number of messages to delete') int count, [
+        @Description('The user from whom to delete messages') User? user,
       ]) async {
-        List<IMessage>? channelMessages;
+        List<Message>? channelMessages;
         Snowflake? last;
 
-        while (count > 0 &&
-            (channelMessages == null || channelMessages.isNotEmpty)) {
-          channelMessages = await context.channel
-              .downloadMessages(
-                limit: 100,
-                after: last,
-              )
-              .toList();
+        while (count > 0 && (channelMessages == null || channelMessages.isNotEmpty)) {
+          channelMessages = await context.channel.messages.fetchMany(
+            limit: 100,
+            after: last,
+          );
+
           last = channelMessages.last.id;
 
-          Iterable<IMessage> toRemove;
+          Iterable<Message> toRemove;
 
           if (user == null) {
             toRemove = channelMessages.take(count);
           } else {
-            toRemove = channelMessages
-                .where((message) => message.author.id == user.id);
+            toRemove = channelMessages.where((message) => message.author.id == user.id);
           }
 
-          try {
-            if (toRemove.length == 1) {
-              await toRemove.first.delete();
-            } else {
-              await context.channel.bulkRemoveMessages(toRemove);
-            }
-          } on IHttpResponseError catch (e) {
-            throw CheckedBotException(e.message);
+          if (toRemove.length == 1) {
+            await toRemove.first.delete();
+          } else {
+            await context.channel.messages.bulkDelete(toRemove.map((m) => m.id));
           }
```

Comment:
This could be reimplemented using [`channel.messages.stream()`](https://github.com/nyxx-discord/nyxx_extensions/blob/dev/lib/src/extensions/managers/message_manager.dart#L23) from `nyxx_extensions` and an `await for` loop - would be a lot cleaner.

Also, we need to handle the case of messages beyond the bulk delete limit.

A simple mock could look something like this:
```dart
final messagesToDelete = await context.channel.messages.stream().where((m) => user == null || user.id == m.author.id).take(count).toList();

await Future.wait(
  messagesToDelete
    .where((m) => m.id.isBefore(Snowflake.firstBulk()))
    .map((m) => m.id)
    .slices(200) // from package:collection
    .map((m) => context.channel.messages.bulkDelete),
);

await Future.wait(
  messagesToDelete
    .where((m) => !m.id.isBefore(Snowflake.firstBulk()))
    .map((m) => m.delete()),
);
```

Reply:
good